# Clustering

Clustering takes data (continuous or quasi-continuous) and adds to them a new categorical group variable that can oftern simplify decision making, even if this sometimes comes at the cost of ignoring intermediate states.

Clustering algorithms are designed to find clusters, so they will find clusters even where there are none.

It is an unsupervised mothod. All variables has the same status. 

Clustering related packages on Bioconductor:
<https://www.bioconductor.org/packages/release/BiocViews.html#___Clustering>

Steps of clustering:

1. observation-by-features rectangular table

2. choose an observation-to observation distance measure

3. compute the distance matrix

4. consturct the cluster eather by agglomerative or partitioning method

### Distance

How to measure similarity?

First, choose the relevant features!

1. Euclidean method: Euclidean distance between two points in a p-dimensional space is the square root of the sum of square of the difference in all p cordinate direcion.

2. Manhattan method (The Manhattan, City Block, Taxicab): Sum of the absolute differences in all coordinates.

3. Maximum method: The maximum of the absolute differences between cordinates.

4. Weighted Euclidean distance: 

5. Minkowski: Allowing the exponent to be m instead of 2, as in the Euclidean distance.

6. Edit, Hamming methods: To compare character sequences like nucleotide or amino acid sequences.

7. Binary:

8. Jaccard distance: co-occurrance

9. Corrlation-based ditance:

Distaces can be computed between any pairs of objects, nut just points in Rp or character sequences. For exapmle the shortest.paths() function (igraph) computes the distance between vertices on a graph, and the function cophenetic computes the distance between leaves of a tree.

Equal-distance contour plots?

# Computations related to distacne in R

dist() function in base R

otions: euclidean, maximum, manhattan, camberra, binary, minkowski

```{r message = F}
library(tidyverse)

data("iris")
iris%>% 
  dplyr::select(1:4)%>%
  dplyr::slice(1:8)%>%
  scale()%>%
  dist()



```

```{r}

iris%>%
  dplyr::select(1:4)%>%
  dplyr::slice(1:8)%>%
  scale()%>%
  dist(method = 'manhattan')

```

Graph-based clustering: single-cell mRNA-sequencing

### k-methodes: k-means, k-medoids and PAM

PAM: partitioning around medioids

partitioning or iterative relocation methodes

steps of the PAM method:
1. take a matrix of p features measured on a set of n observations
2. randomly pick k distinct cluster centers out of the n observations ('seeds')
3. assign each of the remaining observation to the group to whose center is the closest
4. choose a new center from the observation in the group, such that the sum of the distance of group members to the center is minimal, this is called the medoid
5. repeat step 3 and 4 until the groups stabilize

k-means is different from the PAM in the sep 4. The medoid is replaced by the arithmetic means.

Most common off-the-shelf methodes. 
better on: convex, comparable size clusters
different size: biger will be broken up!!

### Resampling

package clusterExperiment: good to scRNA-seq



Challenges in unsupervised clustering of single-cell RNA-seq data PMID: 30617341

community-detection-based algorithms


### Hierarchical clustering
It is a bottom-up approach. (top-down approach: recursive partition methods)
Iterative.

Distance between newly formed clusters:

1. Single linkage (minimal jump) method: It computes the distance between clusters as the smallest distance between any two points in the clusters.

2. Complete linkage (maximum jump) method: It computes the distamce between clusters as the longest distance between any two points in the clusters.

3. Average linkage method: 

4. Centroid method: 

5. Ward`s method: The goal is to minimize the variance within clusters.

```{r}
library(ggplot2)
library(ggdendro) #ggolot2 extension

dm <- dist(USArrests, method = 'euclidean') #calculate distance matrix between points

hc <- hclust(dm, method = "ave") #calculate hierarchical clustering

ggdendrogram(hc) #plot clustering, ggdendrogram is an extended version of base r plot function
```

Common biological application:
Gene expression heatmap.

Look at *pheatmap::pheatmap()* funcion:

1. clustering_distance_rows = "euclidean" - dist()

2. clustering_distance_cols = "euclidean" - dist()

3. clustering_method = "complete" - hclust()


### Validating and choosing the number of clusters

Clustering methods will always deliver groups, even if there is none. 

WSS: within-group sum of square
